ui/components/ui/add-network/add-network.js

```diff
@@ -0,0 +1,146 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { I18nContext } from '../../../contexts/i18n';
+import ActionableMessage from '../actionable-message';
+import Box from '../box';
+import Typography from '../typography';
+import {
+  ALIGN_ITEMS,
+  BLOCK_SIZES,
+  COLORS,
+  DISPLAY,
+  FLEX_DIRECTION,
+  TYPOGRAPHY,
+} from '../../../helpers/constants/design-system';
+import Button from '../button';
+
+const AddNetwork = ({
+  onBackClick,
+  onAddNetworkClick,
+  onAddNetworkManuallyClick,
+  featuredRPCS,
+}) => {
+  const t = useContext(I18nContext);
+
+  const onBack = () => {
+    onBackClick();
+  };
+
+  const onAddNetwork = () => {
+    onAddNetworkClick();
+  };
+
+  const onAddNetworkManually = () => {
+    onAddNetworkManuallyClick();
+  };
+
+  const nets = featuredRPCS
+    .sort((a, b) => (a.ticker > b.ticker ? 1 : -1))
+    .slice(0, 5);
+
+  return (
+    <Box className="add-network__wrapper">
+      <Box
+        height={BLOCK_SIZES.TWO_TWELFTHS}
+        padding={[4, 0, 4, 0]}
+        display={DISPLAY.FLEX}
+        alignItems={ALIGN_ITEMS.CENTER}
+        flexDirection={FLEX_DIRECTION.ROW}
+        className="add-network__header"
+      >
+        <img
+          src="./images/caret-left-black.svg"
+          alt=""
+          onClick={onBack}
+          className="add-network__header__back-icon"
+        />
+        <Typography variant={TYPOGRAPHY.H3} color={COLORS.BLACK}>
+          {t('addNetwork')}
+        </Typography>
+      </Box>
+      <Box
+        height={BLOCK_SIZES.FOUR_FIFTHS}
+        width={BLOCK_SIZES.TEN_TWELFTHS}
+        margin={[0, 6, 0, 6]}
+      >
+        <Typography
+          variant={TYPOGRAPHY.H6}
+          color={COLORS.UI4}
+          margin={[4, 0, 0, 0]}
+        >
+          {t('addFromAListOfPopularNetworks')}
+        </Typography>
+        <Typography
+          variant={TYPOGRAPHY.H7}
+          color={COLORS.UI3}
+          margin={[4, 0, 3, 0]}
+        >
+          {t('customNetworks')}
+        </Typography>
+        {nets.map((item, index) => (
+          <Box
+            key={index}
+            display={DISPLAY.FLEX}
+            alignItems={ALIGN_ITEMS.CENTER}
+            marginBottom={6}
+          >
+            <img
+              className="add-network__token-image"
+              src={item.rpcPrefs.imageUrl}
```

Comment:
could `rpcPrefs` be `undefined` ever?

Reply:
You've got a point.
Fixed in: 124faeca508c9f60b88d0248712309e37a6441e7